### ISSUE TITLE
BCEWithLogitsLoss (Binary) Classification Support?

### ISSUE THREAD

**Issue by spragud2** [AlexImmer/Laplace#107]
Hi,

I have a binary classifier that I am trying to get a last-layer posterior approximation of. The model is trained with BCELossWithLogits, and therefore outputs scalar logits rather than a tensor of length 2.

I have been unable to get la.fit to work with this setup. Changing to a CrossEntropyLoss with 2 classes allows it to work, but the model is large and cumbersome to retrain.

**Comment by wiseodd** [comment]
Unfortunately, it's not supported yet because the Hessian-computation libraries we use (BackPACK and ASDL) do not support BCE yet. Once they've added support to BCE loss, it'll be easy to add it to `laplace-torch`.
